internal/util/merge/merge3.go

```diff
@@ -302,6 +306,7 @@ func (r *resourceHandler) Handle(origin, upstream, local *yaml.RNode) (filters.R
 			r.keptResources = append(r.keptResources, local)
 			strategy = filters.KeepDest
 		}
+	// Do not re-add if deleted from local.
```

Comment:
What is upstream changes from origin and local is deleted ?

Reply:
If the resource has been deleted from local we don't re-add it, even if it is updated in upstream. I think that is a reasonable approach here. The more conservative approach for changes in local is to avoid wiping out consumer changes.